packages/dcos-integration-test/extra/test_networking.py

```diff
@@ -289,3 +288,175 @@ def thread_request():
+    caveat that it doesn't have any default routes. `spartan-net` has routes
+    only for the agent network. In other words it doesn't have any routes
+    towards the spartan-interfaces or minuteman VIPs.
+
+    We then run a server (our python ping-pong server) on the DC/OS overlay.
+    Finally to test that the `dcos-l4lb` plugin, which is also part of
+    `spartan-net` is able to inject the Minuteman and Spartan services into the
+    contianer's netns, we start a client on the `spartan-net` and try to `curl` the
+    `ping-pong` server using its VIP. Without the Minuteman and Spartan services
+    injected in the container's netns the expectation would be that this `curl`
+    would fail, with a successful `curl` execution on the VIP allowing the
+    test-case to PASS.
+    '''
+
+    # CNI configuration of `spartan-net`.
+    spartan_net = {
+        'cniVersion': '0.2.0',
+        'name': 'spartan-net',
+        'type': 'dcos-l4lb',
+        'delegate': {
+            'type': 'mesos-cni-port-mapper',
+            'excludeDevices': ['sprt-cni0'],
+            'chain': 'spartan-net',
+            'delegate': {
+                'type': 'bridge',
+                'bridge': 'sprt-cni0',
+                'ipMasq': True,
+                'isGateway': True,
+                'ipam': {
+                    'type': 'host-local',
+                    'subnet': '192.168.250.0/24',
+                    'routes': [
+                     # Reachability to DC/OS overlay.
+                     {'dst': '9.0.0.0/8'},
+                     # Reachability to all private address subnet. We need
+                     # this reachability since different cloud providers use
+                     # different private address spaces to launch tenant
+                     # networks.
+                     {'dst': '10.0.0.0/8'},
+                     {'dst': '172.16.0.0/12'},
+                     {'dst': '192.168.0.0/16'}
+                    ]
+                }
+            }
+        }
+    }
+
+    log.info("spartan-net config:{}".format(json.dumps(spartan_net)))
+
+    # Application to deploy CNI configuration.
+    cni_config_app, config_uuid = test_helpers.marathon_test_app()
+
+    # Override the default test app command with a command to write the CNI
+    # configuration.
+    #
+    # NOTE: We add the sleep at the end of this command so that the task stays
+    # alive for the test harness to make sure that the task got deployed.
+    # Ideally we should be able to deploy one of tasks using the test harness
+    # but that doesn't seem to be the case here.
+    cni_config_app['cmd'] = 'echo \'{}\' > /opt/mesosphere/etc/dcos/network/cni/spartan.cni; sleep 10000'.format(
+        json.dumps(spartan_net))
+
+    log.info("App for setting CNI config: {}".format(json.dumps(cni_config_app)))
+
+    try:
+        dcos_api_session.marathon.deploy_app(cni_config_app, check_health=False)
```

Comment:
@asridharan Could you please cleanup your containers? I know `conftest.py` will cleanup them for you, but it's not accurate.

You can use the following functions:

* `dcos_api_session.marathon.deploy_and_cleanup(app)`
* `dcos_api_session.marathon.delete('v2/apps/{}'.format(app['id'])).raise_for_status()`

Reply:
Had a chat with @urbanserj and I felt that we shouldn't be using the `deploy_and_cleanup` method or explicit deletes unless we explicitly need to cleanup a container during a tests (as with `test_vips`). 

@mellenburg could chime in as well to give your opinion on this subject?